stdlib/filename.mli

```diff
@@ -49,13 +49,42 @@ val chop_suffix : string -> string -> string
    the filename [name]. The behavior is undefined if [name] does not
    end with the suffix [suff]. *)
 
+val extension : string -> string
+(** [extension name] is the shortest suffix [ext] of [name0] where:
+
+    - [name0] is the longest suffix of [name] that does not
+      contain a directory separator;
+    - [ext] starts with a period;
+    - [ext] is preceded by at least one non-period character
+      in [name0].
+
+    If such a suffix does not exist, [extension name] is the empty
+    string.
+
+    @since 4.04
+*)
+
+
+
 val chop_extension : string -> string
-(** Return the given file name without its extension. The extension
-   is the shortest suffix starting with a period and not including
-   a directory separator, [.xyz] for instance.
+(** Return the given file name without its extension, as defined
+    in {!Filename.extension}.
+
+    Raise [Invalid_argument] if the given name has an empty extension. *)
+
+val chop_extension_if_any : string -> string
```

Comment:
If I guess well this name is introduced for backward compatibility reasons. In that case given that `chop_extension` is really broken from a social convention point of view (`Invalid_argument` is for programming errors). I'd rather `@deprecate` `chop_extension` in favor of this one and give this one a less long winded, odd-looking, name like `remove_extension`.